tests/serverpod_test_server/lib/src/generated/future_calls.dart

```diff
@@ -0,0 +1,160 @@
+/* AUTOMATICALLY GENERATED CODE DO NOT MODIFY */
+/*   To generate run: "serverpod generate"    */
+
+// ignore_for_file: implementation_imports
+// ignore_for_file: library_private_types_in_public_api
+// ignore_for_file: non_constant_identifier_names
+// ignore_for_file: public_member_api_docs
+// ignore_for_file: type_literal_in_constant_pattern
+// ignore_for_file: use_super_parameters
+// ignore_for_file: invalid_use_of_internal_member
+
+// ignore_for_file: no_leading_underscores_for_library_prefixes
+import 'package:serverpod/serverpod.dart' as _i1;
+import 'package:serverpod_test_server/src/generated/simple_data.dart' as _i2;
+import 'dart:async' as _i3;
+import '../futureCalls/test_call.dart' as _i4;
+import '../futureCalls/test_exception_call.dart' as _i5;
+
+/// Invokes a future call.
+typedef _InvokeFutureCall =
+    Future<void> Function(String name, _i1.SerializableModel? object);
+
+/// Global variable for accessing future calls via a typed interface.
+final futureCalls = _FutureCalls();
+
+class _FutureCalls extends _i1.FutureCallInitializer {
+  _i1.FutureCallManager? _futureCallManager;
```

Comment:
Discussion: To follow the same pattern we have when calling `Endpoints` and `Protocol`, it might be better to remove the global variable `futureCalls` in favor of a `FutureCalls()` singleton. Then we expose this singleton under the `futureCalls` getter on the `Serverpod` class, which is from where we expect users to reach the interface.

```suggestion
class FutureCalls extends _i1.FutureCallInitializer {
  FutureCalls._();

  factory FutureCalls() => _instance;

  static final FutureCalls _instance = FutureCalls._();

  _i1.FutureCallManager? __futureCallManager;
```

Reply:
A possible solution for this without having to make `Serverpod` generic is to remove `futureCalls` completely from the `Serverpod` class and generate an extension class on the `future_calls.dart` generated file that declares the getter. This would make the access to the `futureCalls` getter type-safe with no need for casting or generics. In case no `FutureCall` class was generated, the getter won't even be available. Then, we might not even need [the proposed solution of having the `futureCalls` getter on the `EndpointDispatch` class](https://github.com/serverpod/serverpod/pull/4375#discussion_r2644162291) that we were discussing.